airflow/executors/base_executor.py

```diff
@@ -139,6 +148,10 @@ def __repr__(self):
     def start(self):  # pragma: no cover
         """Executors may need to get things started."""
 
+    def log_task_event(self, *, event: str, extra: str, ti_key: TaskInstanceKey):
```

Comment:
@vincbeck  just wanted to explain why i changed the signature of this to take these args instead of just taking a log object.

the reason is, as part of this change, Log object is updated to not require `execution_date` on the ti obj it is passed.  but older versions of airflow don't have this update to Log.  so they fail backcompat tests (installing this provider against e.g. airflow 2.7.3)

the easiest way to fix this is to use this signature instead because, if this method is not present in airflow, we just do nothing -- the event is not logged. and if it _is_ present, then it will have the updates to Log as well.

Reply:
Oh I see! Good catch! And thanks for the explanations. Anyway I feel like it is easier for users to use this method using this signature than having to create a `Log` object

Reply:
So double win